bgpd/bgp_vty.c

```diff
@@ -3425,6 +3425,44 @@ DEFUN (no_neighbor_shutdown,
   return peer_flag_unset_vty (vty, argv[idx_peer]->arg, PEER_FLAG_SHUTDOWN);
 }
 
+DEFUN (neighbor_shutdown_msg,
+       neighbor_shutdown_msg_cmd,
+       "neighbor <A.B.C.D|X:X::X:X|WORD> shutdown message MSG...",
+       NEIGHBOR_STR
+       NEIGHBOR_ADDR_STR2
+       "Administratively shut down this neighbor\n"
+       "Add a shutdown message (draft-ietf-idr-shutdown-03)\n"
+       "Shutdown message\n")
+{
+  int idx_peer = 1;
+  struct peer *peer = peer_lookup_vty (vty, argv[idx_peer]->arg);
+  char *message;
+
+  message = argv_concat (argv, argc, 4);
+  peer_tx_shutdown_message_set (peer, message);
+  XFREE (MTYPE_TMP, message);
+
+  return peer_flag_set_vty (vty, argv[idx_peer]->arg, PEER_FLAG_SHUTDOWN);
+}
+
+DEFUN (no_neighbor_shutdown_msg,
+       no_neighbor_shutdown_msg_cmd,
+       "no neighbor <A.B.C.D|X:X::X:X|WORD> shutdown message MSG...",
```

Comment:
Because it is traditional to accept the cut-n-paste `no` version of the command.  This allows that.